automatic_data_generation/data/base_dataset.py

```diff
@@ -218,6 +222,25 @@ def build_data_files(self, dataset_folder,  dataset_size=None, restrict_intent=N
 
         return new_train_path, new_test_path
 
+    def select_none_intents(self, dataset_folder, restrict_intents, none_folder, cosine_threshold):
+        # select none intents according to overlap with original intents
+        selected_none_intents = []
+        non_intents = []
+        def cosine(u, v):
+            return np.dot(u, v) / (np.linalg.norm(u) * np.linalg.norm(v))
+        intent_vectors = self.load_intent_vectors(dataset_folder)
```

Comment:
don't you need to re-compute the vectors on the restricted dataset? you're kind of cheating if you use the embeddings for the whole one ...